Updated docstring with relevant error raise and examples

John-Aradan opened the pull request:
- Added a Type error raise in case of input is not an array-like object.
- Added thorough examples for the working of the function

fspyridakos commented:
Hey, thanks for your contribution, could you also update the docstrings in the array, container and backend instances of this function?

John-Aradan commented:
@fspyridakos 
#12544 

Updates made as requested:

- 
Added docstring for unique_value: array instance
- Added docstring for unique_value: container instance
- Added docstring for unique_value: numpy frontend instance
- Added docstring for unique_value: torch frontend instance
- Added docstring for unique_value: tensorflow frontend instance
- Added docstring for unique_value: jax frontend instance

John-Aradan commented:
@ivy-seed  Please assign a reviewer for my pull request.

fspyridakos reviewed (changes requested):
Looks really good. I must apologize though for not being clear before, the backend variations don't actually need docstrings, they just need the typehints to be correct, which they seem like they already are, could you please remove them? Apologies again

John-Aradan commented on ivy/array/set.py:
```diff
@@ -44,6 +44,65 @@ def unique_counts(self: ivy.Array) -> Tuple[ivy.Array, ivy.Array]:
+        >>> import ivy
+        >>> x = ivy.array([1, 2, 2, 3, 4, 4, 4])
+        >>> y = ivy.unique_values(x)
+        >>> print(y)
```
Its fine 😊 I will get into it right away

John-Aradan replied:
@fspyridakos  I have made the necessary changes as asked for. Requesting for a re-review.

John-Aradan commented:
@fspyridakos I have made the necessary changes as asked for. Requesting for a re-review.

fspyridakos commented:
Looks good, there seems to be some conflicting changes. Please resolve them and then I'll push 😊

John-Aradan commented:
@fspyridakos I have resolved the merge conflict. Please review 😊

fspyridakos commented:
LGTM! Thanks for your contribution!

John-Aradan commented:
Thankyou ☺️ But, in the subtask set.py unique_values is still not marked as complete
